<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>

<script>
    // wrapped into
    // 网络请求：aaa -> 自己处理10行
    // 处理：aaa111（拼接上111） -> 自己处理10行
    // 处理：aaa111222 -> 自己处理10行
    // new Promise((resolve, reject) => {
    //   setTimeout(() => {
    //     resolve('aaa')
    //   },1000)
    // }).then(res => {
    //   // 1.自己处理10行代码
    //   console.log(res, '第一层的10行处理代码');
    //
    //   // 2.对结果进行第一次处理
    //   return new Promise((resolve, reject) => {
    //     resolve(res + '111')
    //   })
    // }).then(res => {
    //   console.log(res, '第二层的10行代码');
    //
    //   return new Promise((resolve, reject) => {
    //     resolve(res + '222')
    //   })
    // }).then((res) => {
    //   console.log(res, '第三层的10行处理代码');
    // })

    // 2.new Promise(resolve => resolve(结果))简写
    new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve('aaa')
      },1000)
    }).then(res => {
      // 1.自己处理10行代码
      console.log(res, '第一层的10行处理代码');

      // 2.对结果进行第一次处理（简写）
      // return Promise.resolve(res + '111')
      return Promise.reject('error message')    //发生错误信息return error时
      // throw 'error message'  //直接手动抛出错误信息
    }).then(res => {
      console.log(res, '第二层的10行代码');

      return Promise.resolve(res + '222')
    }).then((res) => {
      console.log(res, '第三层的10行处理代码');
    }).catch((err) => {
      console.log(err);
    })

    // 3.省略掉Promise.resolve
    // new Promise((resolve, reject) => {
    //   setTimeout(() => {
    //     resolve('aaa')
    //   },1000)
    // }).then(res => {
    //   // 1.自己处理10行代码
    //   console.log(res, '第一层的10行处理代码');
    //
    //   // 2.对结果进行第一次处理（简写）
    //   return res + '111'
    // }).then(res => {
    //   console.log(res, '第二层的10行代码');
    //
    //   return res + '222'
    // }).then((res) => {
    //   console.log(res, '第三层的10行处理代码');
    // })
</script>
</body>
</html>